src/JustSaying/Messaging/IMessageBatchPublisher.cs

```diff
@@ -0,0 +1,11 @@
+﻿using JustSaying.Messaging.Interrogation;
+using JustSaying.Models;
+
+namespace JustSaying.Messaging;
+
+public interface IMessageBatchPublisher : IInterrogable, IStartable
+{
+    Task PublishAsync(IEnumerable<Message> messages, CancellationToken cancellationToken);
+    Task PublishAsync(IEnumerable<Message> messages, PublishBatchMetadata metadata, CancellationToken cancellationToken);
```

Comment:
Asking to start the discussion, should we cover the use-case where a user needs to have custom metadata per message being sent?

@slang25 and I chatted about this earlier and thought it might be best to leave it for now, and add it later if it's specifically requested.